### PR TITLE
Broker builder implementation

### PR DESCRIPTION
**Pull request by mkrutov**

### CODE REVIEW

**Comment by mkrutov** [comment]
still failing, client not merged in upstream repo yet

File: pkg/framework/framework.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ var (
 	Timeout              = time.Second * 600
 	CleanupRetryInterval = time.Second * 1
 	CleanupTimeout       = time.Second * 5
+	MessagingClientsImage = "quay.io/enmasse/systemtests-clients:latest"
```
Comment (fgiorgetti) [inline pkg/framework/framework.go]: This seems to be misplaced here. Should be put somewhere in the pkg/api/client package.

File: pkg/framework/operators/baseoperator.go [hunk 2]
```diff
@@ -52,6 +52,7 @@ type BaseOperator struct {
 	cRoleBinding      rbacv1.ClusterRoleBinding
 	crds              []apiextv1b1.CustomResourceDefinition
 	keepCRD           bool
+	clientImageName	  string
```
Comment (fgiorgetti) [inline pkg/framework/operators/baseoperator.go]: Same here. The baseoperator does not implicitly needs a client image. This has to be handled separately from the baseoperator.

File: pkg/framework/operators/baseoperator.go [hunk 3]
```diff
@@ -61,6 +62,8 @@ type DefinitionStruct struct {
 	Spec       interface{} `json:"spec"`
 }
 
+const ClientImageName = "quay.io/enmasse/systemtests-clients:latest"
```
Comment (fgiorgetti) [inline pkg/framework/operators/baseoperator.go]: Same as others, as I do not think this is the correct place for client image.

File: pkg/framework/operators/baseoperator.go [hunk 4]
```diff
@@ -334,6 +337,27 @@ func (b *BaseOperator) setupDeployment(jsonItem []byte) {
...
+    clientsDeploymentConfig.Spec.Template.Spec.Containers[0].Name="systemtests-clients"
+    clientsDeploymentConfig.Spec.Template.Spec.Containers[0].Image=ClientImageName
+    clientsDeploymentConfig.Spec.Template.Spec.Containers[0].Command[0]="sleep"
+    clientsDeploymentConfig.Spec.Template.Spec.Containers[0].Args[0]="infinity"
+    clientsDeploymentConfig.Spec.Template.Spec.Containers[0].ImagePullPolicy="Always"
+    if _, err := b.kubeClient.AppsV1().Deployments(b.namespace).Create(clientsDeploymentConfig); err != nil {
+		b.errorItemCreate("deployment", err)
+	}
+}
+
```
Comment (fgiorgetti) [inline pkg/framework/operators/baseoperator.go]: same as the others... not part of the baseoperator to me.

File: pkg/framework/operators/supported.go [hunk 5]
```diff
@@ -14,5 +15,11 @@ var (
 			keepCdrs:     true,
 			apiVersion:   "v1alpha1",
 		}},
+		OperatorTypeBroker: &BrokerOperatorBuilder {BaseOperatorBuilder{
+			image: "brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/amq7/amq-broker-operator:0.9",
+			operatorName: "activemq-artemis-operator",
+			keepCdrs: true,
+			apiVersion: "v1alpha1",
+		}},
```
Comment (fgiorgetti) [inline pkg/framework/operators/supported.go]: Are we going to replace this with an upstream image? The idea is that it is not tied to any downstream stuff.

File: pkg/framework/operators/baseoperator.go (hunk 4 above)
Comment (mkrutov) [reply pkg/framework/operators/baseoperator.go]: Alright, I'll move it out. Thanks!

File: pkg/framework/operators/supported.go (hunk 5 above)
Comment (mkrutov) [reply pkg/framework/operators/supported.go]: Ouch, I've missed this.

**Pull request merged**